docs/architecture-decisions/adr002-default-catalog-file-format.md

```diff
@@ -0,0 +1,245 @@
+</table>
+
+The `metadata` root field has the following nested structure.
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Required</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td valign="top"><code>name</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The name of the entity. Names must be unique per kind, within a given namespace.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>namespace</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The `name` of a namespace that the entity belongs to. This field is optional.
+        </p>
+        <p>
+          Namespaces may also be part of the catalog, and are <code>v1</code>/<code>Namespace</code>
+          objects, i.e. not Backstage specific but the same as in Kubernetes.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>description</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          A human readable description of the entity, to be shown in Backstage. May contain
+          markdown.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>labels</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          Labels are key/value pairs of that are attached to the entity, and their use is
+          identical to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/">kubernetes object labels</a>.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>annotations</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          An object with arbitrary non-identifying metadata attached to the entity,
+          identical in use to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/">kubernetes object annotations</a>.
+        </p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+The precise structure of the `spec` root object depends on the `apiVersion` and
+`kind` combination, and some kinds may not even have a `spec` at all. See
+further down in this document for the specification structure of specific kind.
+
+## Entity Specification: backstage.io/v1 Component
+
+| Field        | Value             |
+| ------------ | ----------------- |
+| `apiVersion` | `backstage.io/v1` |
+| `kind`       | `Component`       |
+
+The `spec` object for this kind is as follows:
+
+| Field  | Type   | Required | Description                            |
+| ------ | ------ | -------- | -------------------------------------- |
+| `type` | String | Yes      | The type of component, e.g. `service`. |
+
+TODO: More fields
```

Comment:
Idd :grin:. I'm thinking we could attack this bit by going data-model and API-first? As in how would plugins consume data that users specify in fields here, and how is it stored and transferred to the frontend.

Reply:
I was a bit wary of mentioning an API wire format in the same document as the ingestion file format here, because it might confuse things.

However, for sanity, we very likely want them to be structured very similarly, if not identically.

Reply:
Yeah not saying we should put it in this doc, just a WoW. Such that we don't spend very much time without knowing how many plugins can consume much data, many wow. 🐕